### PR TITLE
core(tracehouse): improved attribution for XHRs + paint/layout/HTML

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Plugs a couple holes in attribution that didn't need sampling profiler. Brings the "Other" category on the Verge down ~90%.

**Related Issues/PRs**
ref #8526

### CODE REVIEW

File: lighthouse-core/audits/bootup-time.js [hunk 1]
```diff
@@ -88,6 +96,10 @@ class BootupTime extends Audit {
     let attributableURL = jsURL || fallbackURL;
     // If we can't find what URL was responsible for this execution, just attribute it to the root page.
     if (!attributableURL || attributableURL === 'about:blank') attributableURL = 'Other';
+    if (attributableURL === 'Other' && task.selfTime > 3) debugger
```
Comment (connorjclark) [inline lighthouse-core/audits/bootup-time.js]: debugger

File: lighthouse-core/gather/driver.js [hunk 2]
```diff
@@ -113,8 +113,9 @@ class Driver {
       // Not mandatory but not used much
       'blink.console',
 
-      // Most the events we need come in on these two
+      // Most the events we need come in on these three
```
Comment (connorjclark) [inline lighthouse-core/gather/driver.js]:
```suggestion
      // Most of the events we need are from these three
```

File: lighthouse-core/lib/tracehouse/main-thread-tasks.js [hunk 3]
```diff
@@ -426,38 +444,100 @@ class MainThreadTasks {
...
       default:
-        taskURLs = stackFrameURLs;
+        taskURLs = [];
         break;
     }
 
     /** @type {string[]} */
     const attributableURLs = Array.from(parentURLs);
-    for (const url of taskURLs) {
+    for (const url of taskURLs.concat(stackFrameURLs)) {
```
Comment (connorjclark) [inline lighthouse-core/lib/tracehouse/main-thread-tasks.js]: is `[...taskURLs, ...stackFrameURLs]` slower?

File: lighthouse-core/audits/bootup-time.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/audits/bootup-time.js]: but it was so helpful :)

File: lighthouse-core/audits/bootup-time.js [hunk 4]
```diff
@@ -34,6 +34,14 @@ const UIStrings = {
 
 const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
 
+// These trace events, when not triggered by a script inside a particular task, are just general Chrome overhead.
+const BROWSER_TASK_NAMES_SET = new Set([
+  'CpuProfiler::StartProfiling',
```
Comment (connorjclark) [inline lighthouse-core/audits/bootup-time.js]: I feel this `StartProfling` might be good in a category by itself. It's not typical Chrome tasks, it's the overhead of profiling the page.

Comment (patrickhulce) [reply lighthouse-core/audits/bootup-time.js]: yeah this felt like a dicey situation so just calling it `Chrome` felt like a decentish balance, really want `Other` to mean `Unknown` maybe we just rename it?

Comment (dgnext) [reply lighthouse-core/audits/bootup-time.js]: Yes

File: lighthouse-core/audits/bootup-time.js [hunk 5]
```diff
@@ -34,6 +34,14 @@ const UIStrings = {
...
+  'V8.GCCompactor',
```
Comment (paulirish) [inline lighthouse-core/audits/bootup-time.js]: IMO GC and `StartProfiling` are quite different things. The developer has some control over the amount of GC happening.

DevTools perf panel also has grouping by URL and when there's no URL it just says "unattributable". I think that makes sense for GC things. And any tracing/profiling overhead stuff i like putting into "Chrome" or "Browser"

Comment (paulirish) [reply lighthouse-core/audits/bootup-time.js]: On second thought I remember seeing traces where there's a BIG GC at the beginning of a trace. Is that a problem you're solving here? If so I'm interested in a repro. I couldn't get one after trying a few things...

File: lighthouse-core/gather/driver.js [hunk 6]
```diff
@@ -113,8 +113,9 @@ class Driver {
...
       'devtools.timeline',
+      'devtools.timeline.async',
```
Comment (paulirish) [inline lighthouse-core/gather/driver.js]: what's this provide?

File: lighthouse-core/lib/tracehouse/main-thread-tasks.js (hunk 3 above)
Comment (paulirish) [reply lighthouse-core/lib/tracehouse/main-thread-tasks.js]: if i remember right, spreading is actually quite a bit faster.
(js microbenchmarks! my favorite. 😢 )

File: lighthouse-core/audits/bootup-time.js (hunk 4 above)
Comment (paulirish) [reply lighthouse-core/audits/bootup-time.js]: after discussion... `Browser` and `Browser GC`

File: lighthouse-core/lib/tracehouse/main-thread-tasks.js (hunk 3 above)
Comment (patrickhulce) [reply lighthouse-core/lib/tracehouse/main-thread-tasks.js]: tis true, ~33% faster :)

https://jsperf.com/spread-vs-concat-vs-push

File: lighthouse-core/audits/bootup-time.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/audits/bootup-time.js]: eh *significant* time is relative,  but I think it was ~8 ms so scaled up around ~30ms? a few more ms could've been a long task though so definitely not great

File: lighthouse-core/audits/bootup-time.js (hunk 5 above)
Comment (patrickhulce) [reply lighthouse-core/audits/bootup-time.js]: Settled this as Browser, Browser GC, and Unattributable, but yeah important to keep in mind this is only for tasks that do not have any other `EvaluateScript`/timer/functioncall work. I *believe* this is mostly due to protocol overhead GC'ing. It's on the smaller end of the spectrum though

File: lighthouse-core/gather/driver.js (hunk 6 above)
Comment (patrickhulce) [reply lighthouse-core/gather/driver.js]: I don't think I actually needed the events it adds for any changes here, but it provides the [flow events](https://cs.chromium.org/search/?q=devtools.timeline.async+-f:out+lang:cc&type=cs) to track task root initiators. We'll need it eventually to tackle #8526, but I'll revert for now.

**Comment by patrickhulce** [comment]
feedback addressed and ready for re-review

File: lighthouse-cli/test/smokehouse/test-definitions/lantern/lantern-expectations.js [hunk 7]
```diff
@@ -75,4 +75,27 @@ module.exports = [
...
+          // Make sure all of the CPU time is reflected in the perf metrics as well.
+          // The scripts stalls for 3 seconds and lantern has a 4x multiplier so 12s minimum.
+          numericValue: '>12000',
+        },
+        'bootup-time': {
+          details: {
+            items: {
+              0: {
+                url: /main-thread-consumer/,
+                scripting: '>9000',
```
Comment (paulirish) [inline lighthouse-cli/test/smokehouse/test-definitions/lantern/lantern-expectations.js]: lol

**Pull request merged**